Refactor and fix multiline text bug

guysalt opened the pull request:
commit 1 - [ConsoleRender: remove redundant parameter from _print_status_bar](https://github.com/magmax/python-inquirer/pull/355/commits/ca606d555bcd9f89e0a91ccef7ba64d95ac7e4df):
- unused parameter


commit 2 - [ConsoleRender: remove redundant code in _print_header](https://github.com/magmax/python-inquirer/pull/355/commits/511003b26071a22ffac21e2084e944aa4ce90dbc):
- the `print_header` printed every time new line and then move up once. i don't know what the point of working like this. i tried to do some different uses and it's seems to be pretty redundant. 


commit 3 - [ConsoleRender: remove redundant round brackets](https://github.com/magmax/python-inquirer/pull/355/commits/f6a9ff90de5401e070137f9e9501a24df973da42):
- just aesthetics commit. as for the blessed documentation, it's look like the right way to call ANSI clearing functionality is without round brackets


commit 4 - [ConsoleRender: remove function _force_initial_column](https://github.com/magmax/python-inquirer/pull/355/commits/44ea1c816476e644b408bd629eebe8c82eed8d32):
- the `_force_initial_column` in the end of the event loop seems to be duplicate as after it we call the `_relocate` function who call the `_force_initial_column` again.
- then, remove function `_force_initial_column` as it's just print `\r` and used once on `_relocate_ function. and add in `_relocate` function print to `\r`.


commit 5 - [ConsoleRender: adding handling of lines bigger than terminal width](https://github.com/magmax/python-inquirer/pull/355/commits/13ef9a743c998a490f85634b5d4ebe3ed7cf51a7):
- add clearing eos (end of screen) in relocate.
- remove `clear_eos` on start before event loop (as it happens right after).
- adding calculating of the position in the `print_str` method


commit 6 - [ConsoleRender: remove redundant terminal.clear_eol from print_line](https://github.com/magmax/python-inquirer/pull/355/commits/d1c09a5a25d419bcbeb88d0f1de75da6f4f8351b):
- again, seems to be redundant as if i'm understanding it right all the time the line will be clear, because we are doing clear_eos on the start, and every line is a new line... (eol = end of line)


commit 7 - [ConsoleRender: make render_factory static](https://github.com/magmax/python-inquirer/pull/355/commits/a0667003e638fc8b5f5515db78ae776eaaf8246c):
- unused parameter self

Cube707 commented:
@guysalt what is the target of this PR? the mutliline renderer doesn't seem to work?

Cube707 commented:
oh, I see now. You only added support for to long messages, not messages containing manual linebreaks

guysalt commented:
@Cube707 

yes...

there is too many issues about this problem, and as you can see blessed had all the tools i need to do this.

the major challenge was to understanding what's going on there, and tbh there is a lot of mess there. that's why the next pr is mostly about refactor the code.

guysalt commented:
#115 #26

Cube707 commented:
Yes I know the issues, but it wouod be great if you added some more context to your commits, either in the commit messages or in the PR introducing theme. Its really hard for me to see why you introduced these changes (why is stuff redundant? why did you make a methode static? Etc.). I didn't write the original code and its really hard for me to gage your reasoning and its impact. So please edit either the commit messages or this PRs info.

As for the fix: its only a partial fix that only applies to lines longer that `terminal.width` and not to lines containing manual `\n` (see #). This doesn't need to be fixed in this PR, but needs to be clearly stated to make the limitations of the current state clear. This defnetly requires a adjustment of the commit message.

Thanks for your hard work on this 👍. You are absolutely right in that its quite the mess. Thats why I never dared to touch it 😅. Defnetly feel free to rework it completly... don't forget tests though

guysalt commented:
After i dived in this code, i definitely can add supporting of line containing manual line feed (`\n`). it's not something i encounter with like the supporting of lines longer than `terminal.width`, but i can do it later.

Regarding the pull request commits, the code appears to be messy and disorganized, so to be honest i didn't understand what the purpose of some of the lines. Anyway the primary goal of these changes is to modify the logic related to controlling the screen and managing the re-render process. And to accomplish this, i removed any unnecessary code after i changed the logic.

i would try now to edit the commit to be more verbal for your CR :)

guysalt commented:
ok so i update the PR first comment to describe each commit reason. i did it instead of editing the commit's messages because i think the commit's messages need just to describe the change and not the reason.

thanks for your time :)

Cube707 commented:
Note: I beleve the idea of using `clear_eol` for every line printed was to ensure it would properly work even in the case `print_line()` is used spereatly from the "normal" reprinting steps. (the function implyes that a line gets printet, so it should ensure a clean result on itself...)

I believe it would make sense to rework it to only use `clear_eol` and not use `clear_eos` at the beginning. That way we could (with this update or later)  rework the renderer to support single, per line updates, which would be cool.

Cube707 reviewed (approved):
LGTO

guysalt commented:
great!
let's go to the next PR :)
